### PR TITLE
Alberta Disability Credit

### LINKED ISSUE #329: Alberta disability credit
reference: https://www.canada.ca/content/dam/cra-arc/formspubs/pbg/td1ab/td1ab-23e.pdf

### PR DESCRIPTION
**Pull request by 7qiwx**
Fixes #329

### CODE REVIEW

File: policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/disability/base.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+description: Alberta provides this amount under the disability credit. 
+values:
+  2022-12-31: 16_201
+metadata:
+  unit: currency-CAD
+  period: year
+  name: Alberta disability credit.
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/disability/base.yaml:7]:
```suggestion
```

File: policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/disability/base.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  label: Alberta disability credit amount.
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/disability/base.yaml:8]:
```suggestion
  label: Alberta disability credit amount
```

File: policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/disability/base.yaml [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+  reference:
+  - title: Alberta personal tax credits return
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/disability/base.yaml:10]: name to file title

File: policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/disability/base.yaml [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+    href: https://www.canada.ca/content/dam/cra-arc/formspubs/pbg/td1ab/td1ab-23e.pdf
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/disability/base.yaml:11]: add pdf page

File: policyengine_canada/variables/gov/provinces/ab/tax/income/credits/disability/ab_disability_tax_credit_eligible.py [hunk 5]
```diff
@@ -0,0 +1,13 @@
+from policyengine_canada.model_api import *
+
+
+class ab_disability_tax_credit_eligible(Variable):
+    value_type = bool
+    entity = Person
+    label = "Alberta disability tax credit eligbility"
```
Comment (PavelMakarchuk) [inline policyengine_canada/variables/gov/provinces/ab/tax/income/credits/disability/ab_disability_tax_credit_eligible.py:7]:
```suggestion
    label = "Eligible for the Alberta disability tax credit"
```

File: policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/disability/base.yaml [hunk 6]
```diff
@@ -0,0 +1,10 @@
+description: Alberta provides this amount under the disability credit. 
+values:
+  2022-01-01: 16_201
+metadata:
+  unit: currency-CAD
+  period: year
+  label: Alberta disability credit amount
+  reference:
+  - title: Alberta disability credit base amount (Page 1 Section 4)
+    href: https://www.canada.ca/content/dam/cra-arc/formspubs/pbg/td1ab/td1ab-23e.pdf
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/disability/base.yaml:10]:
```suggestion
    href: https://www.canada.ca/content/dam/cra-arc/formspubs/pbg/td1ab/td1ab-23e.pdf#page=1
```

File: policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/disability/base.yaml [hunk 7]
```diff
@@ -0,0 +1,10 @@
+description: Alberta provides this amount under the disability credit. 
+values:
+  2022-01-01: 16_201
+metadata:
+  unit: currency-CAD
+  period: year
+  label: Alberta disability credit amount
+  reference:
+  - title: Alberta disability credit base amount (Page 1 Section 4)
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/disability/base.yaml:9]:
```suggestion
  - title: Alberta disability credit base amount - Section 4
```

File: policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/disability/base.yaml [hunk 8]
```diff
@@ -0,0 +1,10 @@
+description: Alberta provides this amount under the disability credit. 
+values:
+  2022-01-01: 16_201
+metadata:
+  unit: currency-CAD
+  period: year
+  label: Alberta disability credit amount
+  reference:
+  - title: Alberta disability credit base amount - Section 4
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/disability/base.yaml:9]: add the Alberta law - income tax law document

File: policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/disability/base.yaml [hunk 9]
```diff
@@ -0,0 +1,12 @@
+description: Alberta provides this amount under the disability credit. 
+values:
+  2022-01-01: 16_201
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/disability/base.yaml:3]:
```suggestion
  2022-01-01: 15_284
  2023-01-01: 16_201
```

File: policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/disability/base.yaml [hunk 10]
```diff
@@ -0,0 +1,12 @@
...
+metadata:
+  unit: currency-CAD
+  period: year
+  label: Alberta disability credit amount
+  reference:
+  - title: Alberta disability credit base amount - Section 4
+    href: https://www.canada.ca/content/dam/cra-arc/formspubs/pbg/td1ab/td1ab-23e.pdf#page=1
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/disability/base.yaml:10]: add this reference: https://www.canada.ca/content/dam/cra-arc/formspubs/pbg/5009-d/5009-d-22e.pdf

File: changelog_entry.yaml
Comment (PavelMakarchuk) [inline changelog_entry.yaml:1]: need to add calculation from: 

https://www.canada.ca/content/dam/cra-arc/formspubs/pbg/5009-d/5009-d-22e.pdf

Comment (PavelMakarchuk) [reply changelog_entry.yaml:1]: create a variable named `childcare_received` in 
policyengine_canada/variables/household/expenses/childcare/

File: policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/disability/eligible_age.yaml [hunk 11]
```diff
@@ -0,0 +1,9 @@
+description: Alberta provides certain disability credit under this age.
+values:
+  2022-01-01: 18
+metadata:
+  unit: year
+  label: Alberta disability credit age eligibility
+  reference:
+  - title: Alberta Disablitiy credit - Page 2
+    href:  https://www.canada.ca/content/dam/cra-arc/formspubs/pbg/5009-d/5009-d-22e.pdf
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/disability/eligible_age.yaml:9]:
```suggestion
    href:  https://www.canada.ca/content/dam/cra-arc/formspubs/pbg/5009-d/5009-d-22e.pdf#page=2
```

Comment (PavelMakarchuk) [reply policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/disability/eligible_age.yaml:9]: all pdf sources in yaml files

File: policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/disability/max_amount_child.yaml [hunk 12]
```diff
@@ -0,0 +1,10 @@
+description: Alberta provides this maximum amount for child under 18 under disability credit.
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/disability/max_amount_child.yaml:1]:
```suggestion
description: Alberta provides this maximum disability credit amount for children under a certain age threshold.
```

File: policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/disability/max_amount_child.yaml [hunk 13]
```diff
@@ -0,0 +1,10 @@
...
+values:
+  2022-01-01: 11_470
+metadata:
+  unit: currency-CAD
+  period: year
+  label: Alberta disability credit max amount
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/disability/max_amount_child.yaml:7]:
```suggestion
  label: Alberta disability credit younger max amount
```

File: policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/disability/max_amount_total.yaml [hunk 14]
```diff
@@ -0,0 +1,10 @@
+description: Alberta provides this total maximum amount under disability credit.
+values:
+  2022-01-01: 26_754
+metadata:
+  unit: currency-CAD
+  period: year
+  label: Alberta disability credit max amount
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/disability/max_amount_total.yaml:7]:
```suggestion
  label: Alberta disability credit max base amount
```

File: policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/disability/threshold.yaml [hunk 15]
```diff
@@ -0,0 +1,11 @@
+
+description: Alberta provides this threshold under disability credit.
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/disability/threshold.yaml:2]: delete extra line (1)

File: policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/disability/threshold.yaml
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/disability/threshold.yaml:1]: rename to income threshold

File: policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/disability/threshold.yaml [hunk 16]
```diff
@@ -0,0 +1,11 @@
...
+values:
+  2022-01-01: 3_128
+metadata:
+  unit: currency-CAD
+  period: year
+  label: Alberta disability credit threshold
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/disability/threshold.yaml:8]:
```suggestion
  label: Alberta disability credit income threshold
```

File: policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/disability/eligible_age.yaml [hunk 17]
```diff
@@ -0,0 +1,9 @@
+description: Alberta provides certain disability credit under this age.
+values:
+  2022-01-01: 18
+metadata:
+  unit: year
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/disability/eligible_age.yaml:5]:
```suggestion
  unit: year
  period: year
```

File: policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/disability/base.yaml [hunk 18]
```diff
@@ -0,0 +1,10 @@
+description: Alberta provides this amount under the disability credit.
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/disability/base.yaml:1]:
```suggestion
description: Alberta provides this base disability credit amount. 
```

File: policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/disability/base.yaml [hunk 19]
```diff
@@ -0,0 +1,10 @@
+description: Alberta provides this amount under the disability credit. 
+values:
+  2022-01-01: 15_284
+metadata:
+  unit: currency-CAD
+  period: year
+  label: Alberta disability credit amount
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/disability/base.yaml:7]:
```suggestion
  label: Alberta disability credit base amount
```

File: policyengine_canada/variables/gov/provinces/ab/tax/income/credits/disability/ab_disability_tax_credit.py [hunk 20]
```diff
@@ -0,0 +1,20 @@
...
+    label = "Alberta disability tax credit"
+    unit = CAD
+    definition_period = YEAR
+    defined_for = "ab_disability_tax_credit_eligible"
+
+    def formula(person, period, parameters):
+        p = parameters(period).gov.provinces.ab.tax.income.credits.disability
+        childcare_received = person("childcare_received", period)
+        # max_amount_child - (childcare_received - threshold)
+        child_credit = (person("age", period) < p.eligible_age) * max(
```
Comment (PavelMakarchuk) [inline policyengine_canada/variables/gov/provinces/ab/tax/income/credits/disability/ab_disability_tax_credit.py:16]: separate the formula out into multiple steps

File: policyengine_canada/variables/gov/provinces/ab/tax/income/credits/disability/ab_disability_tax_credit.py [hunk 21]
```diff
@@ -0,0 +1,20 @@
...
+    entity = Person
+    label = "Alberta disability tax credit"
+    unit = CAD
+    definition_period = YEAR
+    defined_for = "ab_disability_tax_credit_eligible"
+
+    def formula(person, period, parameters):
+        p = parameters(period).gov.provinces.ab.tax.income.credits.disability
+        childcare_received = person("childcare_received", period)
+        # max_amount_child - (childcare_received - threshold)
```
Comment (PavelMakarchuk) [inline policyengine_canada/variables/gov/provinces/ab/tax/income/credits/disability/ab_disability_tax_credit.py:15]: delete extra code if not needed

File: policyengine_canada/variables/gov/provinces/ab/tax/income/credits/disability/ab_disability_tax_credit.py [hunk 22]
```diff
@@ -0,0 +1,20 @@
...
+        child_credit = (person("age", period) < p.eligible_age) * max(
+            0, p.max_amount_child - max(0, childcare_received - p.threshold)
+        )
+
+        return min(p.max_amount_total, p.base + child_credit)
```
Comment (PavelMakarchuk) [inline policyengine_canada/variables/gov/provinces/ab/tax/income/credits/disability/ab_disability_tax_credit.py:20]:
```suggestion
        return min_(p.max_amount_total, p.base + child_credit)
```

File: policyengine_canada/variables/gov/provinces/ab/tax/income/credits/disability/ab_disability_tax_credit.py (hunk 20 above)
Comment (PavelMakarchuk) [inline policyengine_canada/variables/gov/provinces/ab/tax/income/credits/disability/ab_disability_tax_credit.py:16]:
```suggestion
        child_credit = (person("age", period) < p.eligible_age) * max_(
```

File: policyengine_canada/variables/gov/provinces/ab/tax/income/credits/disability/ab_disability_tax_credit.py [hunk 23]
```diff
@@ -0,0 +1,20 @@
...
+    unit = CAD
+    definition_period = YEAR
+    defined_for = "ab_disability_tax_credit_eligible"
+
+    def formula(person, period, parameters):
+        p = parameters(period).gov.provinces.ab.tax.income.credits.disability
+        childcare_received = person("childcare_received", period)
+        # max_amount_child - (childcare_received - threshold)
+        child_credit = (person("age", period) < p.eligible_age) * max(
+            0, p.max_amount_child - max(0, childcare_received - p.threshold)
```
Comment (PavelMakarchuk) [inline policyengine_canada/variables/gov/provinces/ab/tax/income/credits/disability/ab_disability_tax_credit.py:17]:
```suggestion
            0, p.max_amount_child - max_(0, childcare_received - p.threshold)
```

File: policyengine_canada/tests/gov/provinces/ab/tax/income/credits/disability/ab_disability_tax_credit_eligible.yaml [hunk 24]
```diff
@@ -0,0 +1,17 @@
...
+
+- name: Uneligible person without disability
+  period: 2022
+  input:
+    is_disabled: false
+    province_code: AB
+  output:
+    ab_disability_tax_credit_eligible: false
+
+
```
Comment (PavelMakarchuk) [inline policyengine_canada/tests/gov/provinces/ab/tax/income/credits/disability/ab_disability_tax_credit_eligible.yaml:17]: delete extra lines

File: policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/disability/eligible_age.yaml [hunk 25]
```diff
@@ -0,0 +1,10 @@
+description: Alberta provides an additional disability credit to filers below this age
+values:
+  2022-01-01: 18
+metadata:
+  unit: year
+  period: year
+  label: Alberta disability credit age eligibility
+  reference:
+  - title: Alberta Disablitiy credit - Page 2
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/disability/eligible_age.yaml:9]: try to find this program in the Alberta income tax act - and leave a comment if the program or parts of it are not represented there

File: policyengine_canada/variables/gov/provinces/ab/tax/income/credits/disability/ab_disability_tax_credit.py [hunk 26]
```diff
@@ -0,0 +1,22 @@
...
+    entity = Person
+    label = "Alberta disability tax credit"
+    unit = CAD
+    definition_period = YEAR
+    defined_for = "ab_disability_tax_credit_eligible"
+
+    def formula(person, period, parameters):
+        p = parameters(period).gov.provinces.ab.tax.income.credits.disability
+        childcare_received = person("childcare_received", period)
+        childcare = max_(0, childcare_received - p.income_threshold)
```
Comment (PavelMakarchuk) [inline policyengine_canada/variables/gov/provinces/ab/tax/income/credits/disability/ab_disability_tax_credit.py:15]:
```suggestion
        additional_amount_reduction = max_(0, childcare_received - p.income_threshold)
```

File: policyengine_canada/variables/gov/provinces/ab/tax/income/credits/disability/ab_disability_tax_credit.py [hunk 27]
```diff
@@ -0,0 +1,22 @@
...
+        child_credit_eligibility = person("age", period) < p.eligible_age
```
Comment (PavelMakarchuk) [inline policyengine_canada/variables/gov/provinces/ab/tax/income/credits/disability/ab_disability_tax_credit.py:16]:
```suggestion
        additional_amount_eligibility = person("age", period) < p.eligible_age
```

File: policyengine_canada/variables/gov/provinces/ab/tax/income/credits/disability/ab_disability_tax_credit.py [hunk 28]
```diff
@@ -0,0 +1,22 @@
...
+        child_credit = (
```
Comment (PavelMakarchuk) [inline policyengine_canada/variables/gov/provinces/ab/tax/income/credits/disability/ab_disability_tax_credit.py:17]:
```suggestion
        max_additional_amount =  max_(0, p.max_amount.younger - childcare)
        additional_amount = where(additional_amount_eligibile, max_additional_amount, 0)
```

File: policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/disability/eligible_age.yaml [hunk 29]
```diff
@@ -0,0 +1,10 @@
+description: Alberta provides an additional disability credit to filers below this age
+values:
+  2022-01-01: 18
+metadata:
+  unit: year
+  period: year
+  label: Alberta disability credit age eligibility
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/disability/eligible_age.yaml:7]:
```suggestion
  label: Alberta disability credit additional amount age eligibility
```

File: policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/disability/income_threshold.yaml [hunk 30]
```diff
@@ -0,0 +1,10 @@
+description: Alberta provides this threshold under disability credit.
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/disability/income_threshold.yaml:1]:
```suggestion
description: Alberta reduces the additional disability credit amount by the excess of childcare expenses over this threshold.
```

File: policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/disability/eligible_age.yaml
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/disability/eligible_age.yaml:1]: make a folder called `additional_amount` - everything but base

File: policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/disability/max_amount_total.yaml [hunk 31]
```diff
@@ -0,0 +1,10 @@
+description: Alberta limits the disability credit to this maximum base amount.
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/disability/max_amount_total.yaml:1]:
```suggestion
description: Alberta limits the disability credit to this maximum base amount.
```

File: policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/disability/max_amount_total.yaml [hunk 32]
```diff
@@ -0,0 +1,10 @@
...
+values:
+  2022-01-01: 26_754
+metadata:
+  unit: currency-CAD
+  period: year
+  label: Alberta disability credit max base amount
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/disability/max_amount_total.yaml:7]:
```suggestion
  label: Alberta disability credit max amount
```

File: policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/disability/max_amount/younger.yaml [hunk 33]
```diff
@@ -0,0 +1,10 @@
+description: Alberta provides this maximum disability credit amount for children under a certain age threshold.
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/disability/max_amount/younger.yaml:1]:
```suggestion
description: Alberta provides this additional disability credit amount for younger filers.
```

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_canada/variables/gov/provinces/on/tax/income/credits/oeptc/oeptc_category.py [hunk 34]
```diff
@@ -14,6 +14,7 @@ class oeptc_category(Variable):
     possible_values = OEPTCCategory
     default_value = OEPTCCategory.SINGLE_SHARED_CUSTODY
     definition_period = YEAR
+    label = "oeptc category"
```
Comment (MaxGhenis) [inline policyengine_canada/variables/gov/provinces/on/tax/income/credits/oeptc/oeptc_category.py:17]:
```suggestion
    label = "OEPTC category"
```

File: policyengine_canada/variables/gov/provinces/ab/tax/income/credits/disability/ab_disability_credit.py [hunk 35]
```diff
@@ -0,0 +1,19 @@
...
+    definition_period = YEAR
+    defined_for = "ab_disability_credit_eligible"
+    reference = "https://www.canada.ca/content/dam/cra-arc/formspubs/pbg/5009-d/5009-d-22e.pdf#page=2"
+
+    def formula(person, period, parameters):
+        p = parameters(period).gov.provinces.ab.tax.income.credits.disability
+        additional_amount = person(
+            "ab_disability_credit_additional_amount", period
+        )
+        total_credit_amount = p.base + additional_amount
```
Comment (MaxGhenis) [inline policyengine_canada/variables/gov/provinces/ab/tax/income/credits/disability/ab_disability_credit.py:18]:
```suggestion
        uncapped_amount = p.base + additional_amount
```

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

File: policyengine_canada/variables/household/person/care_expenses.py [hunk 36]
```diff
@@ -0,0 +1,9 @@
+from policyengine_canada.model_api import *
+
+
+class care_expenses(Variable):
+    value_type = float
+    entity = Person
+    label = "Amount of childcare expenses incurred by the child"
```
Comment (MaxGhenis) [inline policyengine_canada/variables/household/person/care_expenses.py:7]: So we can include elder and disability care in it
```suggestion
    label = "Amount of care expenses incurred by the person"
```

File: policyengine_canada/variables/gov/provinces/ab/tax/income/credits/disability/ab_disability_credit_additional_amount.py [hunk 37]
```diff
@@ -0,0 +1,23 @@
...
+    label = "Alberta additional disability tax credit"
+    unit = CAD
+    definition_period = YEAR
+    defined_for = ProvinceCode.AB
+    reference = "https://www.canada.ca/content/dam/cra-arc/formspubs/pbg/5009-d/5009-d-22e.pdf#page=2"
+
+    def formula(person, period, parameters):
+        p = parameters(period).gov.provinces.ab.tax.income.credits.disability
+        childcare_expenses = person("care_expenses", period)
+        reduced_childcare_expenses = max_(
```
Comment (MaxGhenis) [inline policyengine_canada/variables/gov/provinces/ab/tax/income/credits/disability/ab_disability_credit_additional_amount.py:16]: More conventional term
```suggestion
        excess_childcare_expenses = max_(
```

**Review by MaxGhenis — CHANGES_REQUESTED** [review]
can we hook this up to the net income tree?

File: changelog_entry.yaml
Comment (PavelMakarchuk) [inline changelog_entry.yaml:1]: could you create an ab_refundable / ab_non_refundable credits python and respective refundable.yaml and non_refundable.yaml files 
This is the income tree structure which we need to integrate the program

Comment (PavelMakarchuk) [reply changelog_entry.yaml:1]: add the credit which we have created here to the right parameter file

File: policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/non_refundable.yaml [hunk 38]
```diff
@@ -0,0 +1,10 @@
+description: Alberta list of non-refundable tax credits.
+values:
+  2022-01-01:
+    - ab_age_credit
+    - ab_disability_credit
+    - ab_pension_credit
+    - ab_spouse_and_common_law_partner_amount_credit
+
+metadata:
+  unit: program
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/non_refundable.yaml:10]:
```suggestion
  unit: list
  period: year 
  label: Alberta non-refundable credits
```

File: policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/non_refundable.yaml [hunk 39]
```diff
@@ -0,0 +1,10 @@
+description: Alberta list of non-refundable tax credits.
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/non_refundable.yaml:1]:
```suggestion
description: Alberta provides the follwoing non-refundable tax credits.
```

File: policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/non_refundable.yaml (hunk 38 above)
Comment (PavelMakarchuk) [reply policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/non_refundable.yaml:10]: also add reference (tax form and law) - general for the credits

File: policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/refundable.yaml [hunk 40]
```diff
@@ -0,0 +1,6 @@
+description: Alberta list of refundable tax credits.
+values:
+  2022-01-01:
+
+metadata:
+  unit: program
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/refundable.yaml:6]: same adjustments as in non-ref.

File: policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/refundable.yaml [hunk 41]
```diff
@@ -0,0 +1,6 @@
+description: Alberta list of refundable tax credits.
+values:
+  2022-01-01:
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/refundable.yaml:3]:
```suggestion
  2022-01-01: []
```

File: policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/refundable.yaml [hunk 42]
```diff
@@ -0,0 +1,6 @@
+description: Alberta list of refundable tax credits.
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/refundable.yaml:1]: same as non-ref.

File: policyengine_canada/variables/gov/provinces/ab/tax/income/credits/ab_non_refundable_credits.py [hunk 43]
```diff
@@ -0,0 +1,11 @@
+from policyengine_canada.model_api import *
+
+
+class ab_non_refundable_credits(Variable):
+    value_type = float
+    entity = Person
```
Comment (PavelMakarchuk) [inline policyengine_canada/variables/gov/provinces/ab/tax/income/credits/ab_non_refundable_credits.py:6]:
```suggestion
    entity = Household
```

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/non_refundable.yaml [hunk 44]
```diff
@@ -0,0 +1,17 @@
+description: Alberta provides the follwoing non-refundable tax credits.
```
Comment (MaxGhenis) [inline policyengine_canada/parameters/gov/provinces/ab/tax/income/credits/non_refundable.yaml:1]:
```suggestion
description: Alberta provides the following non-refundable tax credits.  
```

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**